src/network/networkd-nexthop.c

```diff
@@ -834,10 +893,54 @@ int manager_rtnl_process_nexthop(sd_netlink *rtnl, sd_netlink_message *message,
                 return 0;
         }
 
-        r = netlink_message_read_in_addr_union(message, NHA_GATEWAY, tmp->family, &tmp->gw);
+        r = sd_netlink_message_read_data(message, NHA_GROUP, &raw_group_size, &raw_group);
         if (r < 0 && r != -ENODATA) {
-                log_link_warning_errno(link, r, "rtnl: could not get NHA_GATEWAY attribute, ignoring: %m");
+                log_link_warning_errno(link, r, "rtnl: could not get NHA_GROUP attribute, ignoring: %m");
                 return 0;
+        } else if (r >= 0) {
+                struct nexthop_grp *group = raw_group;
```

Comment:
Do we know that raw_group is aligned? On amd64 this will not cause a problem, but it might on other architectures.
Maybe add `assert(group % __alignof__(struct nexthop_grp) == 0);` ?